include/net/net_pkt.h

```diff
@@ -1346,6 +1346,19 @@ void net_pkt_get_info(struct k_mem_slab **rx,
 		      struct net_buf_pool **rx_data,
 		      struct net_buf_pool **tx_data);
 
+/**
+ * @brief Get source socket address.
+ *
+ * @param pkt Nework packet
+ * @param addr Source socket address
+ * @param addrlen The length of source socket address
```

Comment:
Once semantics of *addrlen figure out (comment below), this docstring should describe what gets there on input, what on output.